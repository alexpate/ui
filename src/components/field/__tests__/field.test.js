/* eslint-env jest */
import expect from 'expect';
import {shallow} from 'enzyme';
import React from 'react';

import Input from 'components/input';

import Field from '../field';

describe('components/field', () => {
  describe('Field (default)', () => {
    it('should render as expected', () => {
      const component = (
        <Field label="My Label" hint="My Hint">
          <Input value="My Input Value" />
        </Field>
      );
      const result = shallow(component);
      expect(result.html()).toMatch(/My Label/);
      expect(result.html()).toMatch(/My Hint/);
      expect(result.find(Input).is({value: 'My Input Value'})).toBe(true);
    });
  });
});
