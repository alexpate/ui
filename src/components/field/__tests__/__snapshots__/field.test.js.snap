// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field should render shallow component ok 1`] = `
<div
  className="ui-field ui-field--legacy"
  onClick={undefined}
  style={Object {}}
>
  <div
    className={null}
    style={
      Object {
        "alignContent": "flex-start",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexShrink": 0,
        "justifyContent": "space-between",
        "position": "relative",
      }
    }
  >
    <label
      className="ui-field-label"
      htmlFor={undefined}
      style={
        Object {
          "marginBottom": "4px",
          "marginRight": 0,
          "width": null,
        }
      }
      title={undefined}
    >
      My Label
    </label>
  </div>
  <div
    className="ui-input-container ui-input-container--margin-none"
  >
    <input
      className="ui-input ui-input--medium"
      defaultValue={undefined}
      disabled={false}
      name={undefined}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder={undefined}
      readOnly={false}
      style={
        Object {
          "width": undefined,
        }
      }
      value="My Input Value"
    />
  </div>
  <span
    className="ui-field-hint"
    style={Object {}}
  >
    My Hint
  </span>
</div>
`;
